features/nfc/nfc/NFCEEPROM.h

```diff
@@ -0,0 +1,160 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_NFC_EEPROM_H
+#define MBED_NFC_EEPROM_H
+
+#include <stdint.h>
+#include "events/EventQueue.h"
+
+#include "NFCDefinitions.h"
+#include "NFCTarget.h"
+#include "NFCEEPROMDriver.h"
+
+namespace mbed {
+namespace nfc {
+
+/**
+ * @addtogroup nfc
+ * @{
+ */
+
+/**
+ * The NFC EEPROM class represents a NFC target device connected using a wired
+ * link (I2C, SPI, etc).
+ *
+ * These EEPROMs essentially provide addressable memory that can be accessed
+ * using either a wired or NFC interface.
+ *
+ * In NFC mode these most often conform to one of the NFC tag types defined
+ * by the NFC Forum, therefore encoding NDEF data in these EEPROMs will
+ * ensure that it is understandable by a NFC reader.
+ */
+class NFCEEPROM : public NFCTarget, public NFCEEPROMDriver::Delegate {
```

Comment:
maybe composition would work better here? it's confusing how there are two delegates